pkg/controller/controller.go

```diff
@@ -661,9 +661,9 @@ func (p *csiProvisioner) getPVCSource(options controller.ProvisionOptions) (*csi
 	if sourcePVC.ObjectMeta.DeletionTimestamp != nil {
 		return nil, fmt.Errorf("the PVC DataSource %s is currently being deleted", options.PVC.Spec.DataSource.Name)
 	}
-	if sourcePVC.Spec.StorageClassName != options.PVC.Spec.StorageClassName {
+	if *sourcePVC.Spec.StorageClassName != *options.PVC.Spec.StorageClassName {
```

Comment:
Could this be nil?

Also can we add a unit test for this?

Reply:
@msau42 yes this can be nil also. updated code and added unit testing.

Reply:
DataSource and rquested provisioned PVC are *required* to be in the same storage class.  Also, I'm not sure I follow, how can the sourcePVC have a "nil" storage class?  Once it's provisioned it will have a storage class, even if one wasn't specified at creation, the default class would be used and the object would have a storage class assigned to it no?

Reply:
@j-griffith this is a kind of safety check, will it cause any issue if we do `nil` check?

Reply:
It won't hurt at all, and might as well err on the side of caution.